kv/raftstore/peer_storage.go

```diff
@@ -544,26 +473,15 @@ func (ps *PeerStorage) SaveReadyState(ready *raft.Ready) (*ApplySnapResult, erro
 	kvWB.MustWriteToDB(ps.Engines.Kv)
 	raftWB.MustWriteToDB(ps.Engines.Raft)
 
-	ps.raftState = ctx.RaftState
 	ps.lastTerm = ctx.lastTerm
-
+	ps.raftState = ctx.RaftState
 	// If we apply snapshot ok, we should update some infos like applied index too.
-	if ctx.SnapRegion == nil {
-		return nil, nil
+	if snapshotIdx > 0 {
+		ps.region = ctx.SnapRegion
+		ps.ScheduleApplyingSnapshot()
 	}
-	// cleanup data before scheduling apply worker.Task
-	if ps.isInitialized() {
-		ps.clearExtraData(ps.region)
```

Comment:
why remove this

Reply:
`clearExtraData` is used to clear data that not overlap by `ps.region` and the input region, but in here the input region is `ps.region` too, so they always overlap and `clearExtraData` is useless.

Reply:
after apply snapshot, the region may change

Reply:
oh, got it.